src/core/server/elasticsearch/client/client_facade.mock.ts

```diff
@@ -0,0 +1,423 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ApiResponse } from '@elastic/elasticsearch';
+import { TransportRequestPromise } from '@elastic/elasticsearch/lib/Transport';
+import { ClientFacade } from './client_facade';
+
+const createApiResponse = <T>(body: T): TransportRequestPromise<ApiResponse<T>> => {
+  const response: ApiResponse<T> = {
+    body,
+    statusCode: 200,
+    warnings: [],
+    headers: {},
+    meta: {} as any,
+  };
+  const promise = Promise.resolve(response);
+  (promise as TransportRequestPromise<ApiResponse<T>>).abort = () => undefined;
```

Comment:
A downside of using this extended `TransportRequestPromise` version of promise is that you can't just use `jest.fn().mockResolvedValue`, as there is a signature mismatch between `Promise` and `TransportRequestPromise`. I was kinda forced to introduce these utility methods and to use them as `jest.fn().mockReturnValue(createApiResponse(...))` instead. (see `ensure_es_version.test.ts`)